Copy for the top-level page of the "CHT Applications" (#240)
medic/cht-docs: content/en/apps/_index.md

To create a digital health tool using the Community Health Toolkit, you'll need to build a CHT Application or re-use an existing CHT Application.

CHT Applications have been built to support a range of health interventions including antenatal care, postnatal care, family planning, integrated community case management of childhood illness, malnutrition, immunization, and epidemic response. CHT Applications have been built to support users at all levels of the community health system including patients, CHWs, CHW supervisors, nurses, health facility staff, programme staff, researchers, and policy makers. Beyond the community, data from CHT Applications can integrate with the broader health system through integrations with software such as OpenMRS, DHIS2, and RapidPro.

## What skills are required

Building a CHT Application or altering an existing app is a technical undertaking requiring a technical skillset. 

### To Build an App

> Technically speaking, CHT Applications consist of JSON files, XLSForms, JavaScript code, media (images/videos), and translations. Collectively, this is referred to as "configuration code". The following technical skills are necessary to work with configuration code.

Reviewer: Depending on what you want to do not all skills are _necessary_, so simpler to say _useful_. Open to other ways of saying that though.

```suggestion
Technically speaking, CHT Applications consist of JSON files, XLSForms, JavaScript code, media (images/videos), and translations. Collectively, this is referred to as "configuration code". The following technical skills are useful to work with configuration code.
```

> * XLSForms
> * JavaScript code and JavaScript expressions
> * JSON for specify configuration
> * Use of command line tools

Reviewer: Great to have this as a list! 

The list is presented as "technical skills", but some items list technologies. Attempted update for consistency, and provide a bit more context about when each skill is used:
```suggestion
* Use of command line tools to compile, update, and upload your app
* [XLSForms](http://xlsform.org) for building contact forms and actions
* JSON for modifying settings, tasks, targets
* JavaScript for advanced configuration
```

identifier: "apps"
weight: 2
description: >
  Overview and reference for building CHT applications
---

> To create a digital health tool using the Community Health Toolkit: build a CHT Application or re-use an existing CHT Application. The [CHT Core Framework]({{ ref="core" }}) provides a foundation on which many digital health tools can be conceptualised and built. A CHT Application takes that foundation and builds on it to create a concrete digital health tool tailored to the health workflows in a targeted community. 

Reviewer: I find this paragraph tough to read and get into. We also remove the concept of configuration code, which is something we've used organizationally for a while. I am ok with changing the term, but it is useful as a concept so it would be good to have a larger conversation about what we want to call that.

An attempt to reword, feel free to use as you wish.

```suggestion
CHT applications are digital health apps built using the Core Framework of the Community Health Toolkit. The [CHT Core Framework]({{< ref="core" >}}) provides a foundation on which a concrete digital health app can be built, tailored to the needs of the health workers and the community. This custom building of CHT apps is done in a modular way, bringing in settings, images, and code as needed.
```

Minor note that from what I've seen we typically use "app" over "tool", unless it is a specific aspect within the app. @CescaGarrett might have more on that language.

Author: I thought we renamed "configuration code" as "application code"?

Reviewer: I wasn't aware (or forgot) about that change... I remember the cons of "applicaton code" were that it makes it hard to distinguish with what was previously _non-configuration code_, and I don't remember us renaming that to _framework code_, _core application code_, or something else.

> Since all CHT Applications share the same foundation, they also share capabilities and attributes. For example, all CHT Applications share a similar user-interface and share the same look-and-feel. All CHT Applications can be built to scale, can run across a variety of hardware, support offline-first experiences, and support most languages. These baseline capabilities are universal across all CHT Applications. Learn more about this in [Why the CHT?]({{ ref="why-the-cht" }})

Reviewer: ```suggestion
Since all CHT Applications share the same foundation, they also share capabilities and attributes. For example, all CHT Applications share a similar user-interface and share the same look-and-feel. All CHT Applications can be built to scale, can run across a variety of hardware, support offline-first experiences, and support most languages. These baseline capabilities are universal across all CHT Applications. Learn more about this in [Why the CHT?]({{< ref="why-the-cht" >}})
```

> Although CHT Applications share this common foundation, they can be built to tackle the diverse health challenges of communities around the globe. Some examples of health interventions which have been digitized by CHT applications include antenatal care, postnatal care, integrated community case management, family planning, malnutrition, immunization, and epidemic response. 

Author: So my thinking was:

Paragraph 1-- very roughly, what is a CHT Application? It is a digital health tool following a specific pattern
Paragraph 2 -- what is the framework anyway? What is the relationship between the framework and the app? how are cht apps similar?
Paragraph 3 (this paragraph) --  what sort of things can apps do? (how is each one unique and highly contextualised to a specific problem)

If we are going to jump into the idea of "profiles for families" I think we need to set the stage quite a bit before that. Do readers know that these apps are centered around filling forms, completing workflows, and saving that data? I think contact profiles are too deep to explain right away... So, I'd skip your first sentence.

But I love the people first and some other edits. What do you think about touching on cases where our users are not CHWs or CHW Supervisors (like MoH-Covid-KE or Goma) for example? We focus on CHWs, but the tool isn't exclusive to them. I'm not sure what is the right way to message that, if at all.

> Although CHT Applications share this common foundation, they can be built to tackle the diverse health challenges of communities around the globe. Some examples of health interventions which have been digitized by CHT applications include antenatal care, postnatal care, integrated community case management, family planning, malnutrition, immunization, and epidemic response. 

Reviewer 2: I really like the changes overall, the language around CHT applications and CHT Core as a foundation seems clearer. On this part:

>  CHT apps have been used by CHWs supporting their communities with a variety of health aspects...

I agree with @abbyad's feedback, yet it might benefit from further iteration. Perhaps like:
> Custom CHT applications have been built to support a range of health interventions, including antenatal care, postnatal care, family planning, integrated community case management of childhood illness, malnutrition, immunization, and epidemic response. 

Happy to see this continue to improve!

> Although CHT Applications share this common foundation, they can be built to tackle the diverse health challenges of communities around the globe. Some examples of health interventions which have been digitized by CHT applications include antenatal care, postnatal care, integrated community case management, family planning, malnutrition, immunization, and epidemic response. 

Reviewer 2: > If we are going to jump into the idea of "profiles for families" I think we need to set the stage quite a bit before that. Do readers know that these apps are centered around filling forms, completing workflows, and saving that data? I think contact profiles are too deep to explain right away... So, I'd skip your first sentence and leave that to the top-level paragraphs on the "Features" page.

In the technical docs it makes sense to me to focus first on the technical components and how they come together. The features page and some top level pages seem like good places to focus first on describing health records or task management when an audience is familiar with these feature sets in other digital health software, but not necessarily as concerned with the the technical components of CHT apps. 

> What do you think about touching on cases where our users are not CHWs or CHW Supervisors (like MoH-Covid-KE or Goma) for example? We focus on CHWs, but the tool isn't exclusive to them. I'm not sure what is the right way to message that, if at all and that is why I ended up on the more generic "health tool"

I'm all for describing for the more generic user in our docs, for example we also have CHT applications that are used by a mix of nurses and patients and don't involve any CHWs, and this is still very much under the umbrella of a "community health system". I think we can refer to _health workers_ using _CHT apps/digital health apps/digital tools_, but _health tool_ might be a little too generic.

The CHT Core Framework allows you to define each element in your app in a modular way, and then specify when and how it should appear for different types of users, without having to modify the underlying framework. Collectively, this customization is referred to as Configuration Code.
Although CHT Applications share this common foundation, they can be built to tackle the diverse health challenges of communities around the globe. Some examples of health interventions which have been digitized by CHT applications include antenatal care, postnatal care, integrated community case management, family planning, malnutrition, immunization, and epidemic response. 

Developing an app using the Core Framework requires an understanding of:
- Javascript code and expressions
- JSON format used to specify configuration

> - XLSForms to setup actions and contacts

Reviewer: Where do we first cover this expectation? I think it is fine to remove it if it is covered well elsewhere, just want the basic skills to build apps to be made clear early enough.

Author: Attempted to also capture the skills required to deploy/host the app. Let me know what you think of that addition?

> This section provides an introduction to CHT Applications, provides some sample CHT Applications, and provides reference materials for those wanting to build a new app or deploy an existing app.

Reviewer: ```suggestion
This section provides an introduction to CHT Applications, provides some sample CHT Applications, and provides guides and reference materials for those wanting to build a new app or deploy an existing app.
```

Reviewer: I think this makes a good first paragraph for this page in a "pageinfo" block.

identifier: "apps"
weight: 2
description: >
  Overview and reference for building CHT Applications
---

{{% pageinfo %}}
This section provides an introduction to CHT Applications, provides some sample CHT Applications, and provides guides and reference materials for those wanting to build a CHT Application or deploy an existing one.
{{% /pageinfo %}}

> Community Health Toolkit (CHT) Applications are digital health tools built using the [CHT Core Framework]({{ ref core }}). The Core Framework provides a foundation on which custom CHT Applications are built. Since all CHT Applications share the same foundation, they also share capabilities and attributes. For example, all CHT Applications share a similar user-interface and share the same look-and-feel. All CHT Applications can be built to scale, can run across a variety of devices, support offline-first experiences, and support most languages. These baseline capabilities are foundational across all CHT Applications. Read more about this in [Why the CHT?]({{< ref "/why-the-cht" >}}).r

Reviewer: It's possible the ref shortcode would work without the quotes, but it won't get called without the `<>`. Also removed an errant letter at end of paragraph.

```suggestion
Community Health Toolkit (CHT) Applications are digital health tools built using the [CHT Core Framework]({{< ref "core" >}}). The Core Framework provides a foundation on which custom CHT Applications are built. Since all CHT Applications share the same foundation, they also share capabilities and attributes. For example, all CHT Applications share a similar user-interface and share the same look-and-feel. All CHT Applications can be built to scale, can run across a variety of devices, support offline-first experiences, and support most languages. These baseline capabilities are foundational across all CHT Applications. Read more about this in [Why the CHT?]({{< ref "why-the-cht" >}}).
```

> To create a digital health tool using the Community Health Toolkit, you'll need to build a CHT Application or re-use an existing CHT Application.

Reviewer: I still prefer "digital health app" over "digital health tool", but that may just be me!

> CHT Applications have been built to support a range of health interventions including antenatal care, postnatal care, family planning, integrated community case management of childhood illness, malnutrition, immunization, and epidemic response. CHT Applications have been built to support users at all levels of the community health system including patients, CHWs, CHW supervisors, nurses, health facility staff, programme staff, researchers, and policy makers. Beyond the community, data from CHT Applications can integrate with the broader health system through integrations with software such as OpenMRS, DHIS2, and RapidPro.

Reviewer: Minor edits, and using American English. We can't get away with our Canadian pick and choose approach anymore :)

```suggestion
CHT Applications have been built to support a range of health interventions including antenatal care, postnatal care, family planning, integrated community case management of childhood illness, malnutrition, immunization, and epidemic response. CHT Applications have been built to support users at all levels of the community health system including patients, CHWs, CHW supervisors, nurses, health facility staff, program staff, researchers, and policy makers. Beyond the community-level, data from CHT Applications can integrate with the broader health system through integrations with software such as OpenMRS, DHIS2, and RapidPro.
```
